src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1108,6 +1125,282 @@ bool os::dll_address_to_library_name(address addr, char* buf,
   return true;
 }
 
+// get the library search path burned in to the executable file during linking
+// If the libpath cannot be retrieved return an empty path
```

Comment:
I found a leak in my previous implementation. It is more or less academical, but this solution is the complete one. I would prefer this complete solution even it is complex, because if dlopen follows a slightly different algorithm in resolving the library we surely get into trouble.
If we omit the xcoff32 we have to ensure that no xcoff32 executable file comes into play.

Reply:
>  If we omit the xcoff32 we have to ensure that no xcoff32 executable file comes into play.

xcoff32 is for 32-bit binaries. The AIX port only exists for 64-bit, and there will never be a 32-bit AIX port, so there is no reason for handling 32-bit xcoff headers.